incubator/hnc/pkg/reconcilers/hierarchical_namespace.go

```diff
@@ -210,6 +217,19 @@ func (r *HierarchicalNamespaceReconciler) writeInstance(ctx context.Context, log
 	return nil
 }
 
+func (r *HierarchicalNamespaceReconciler) writeNamespace(ctx context.Context, log logr.Logger, nm, pnm string) error {
+	inst := &corev1.Namespace{}
+	inst.ObjectMeta.Name = nm
+	metadata.SetAnnotation(inst, api.AnnotationOwner, pnm)
+
+	log.Info("Creating namespace on apiserver")
+	if err := r.Create(ctx, inst); err != nil {
```

Comment:
Might be worth leaving a few comments here:
* If the namespace already exists, it means someone else has created a namespace while this reconciler was running, and there's likely a conflict. Returning an error will ensure that this reconciler runs again and is able to report the conflict.
* Any other transient problem should also be retried.

Reply:
sg, will do!